lib/module_3_1/ab_testing_module_3_1.ts

```diff
@@ -17,96 +16,63 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import {
-  Stack, App, StackProps, CfnOutput,
-  aws_s3_deployment as s3deployment,
-  aws_cloudfront_origins as origins,
-  aws_cloudfront as cloudfront,
-  aws_s3 as s3,
-  aws_ssm as ssm,
-  aws_lambda as lambda
-} from "aws-cdk-lib";
-
+import { Stack, App, StackProps, CfnOutput } from "aws-cdk-lib";
+import { Distribution, Function, FunctionCode, FunctionEventType, FunctionRuntime, ViewerProtocolPolicy } from "aws-cdk-lib/aws-cloudfront";
+import { S3Origin } from "aws-cdk-lib/aws-cloudfront-origins";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { join } from "path";
 
-import { ABDashboard } from '../ab_dashboard';
+import { ABDashboard } from "../ab_dashboard";
+import { FunctionWithStore } from "./function-with-store";
 
+const RESOURCES_PATH_PREFIX = join(__dirname, "../../resources/module_3_1");
 
 export class Module_3_1 extends Stack {
-
   constructor(scope: App, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    const hostingBucket = new s3.Bucket(this, 'hosting-bucket-deployment');
-    const configBucket = new s3.Bucket(this, 'config-bucket-deployment');
-
-    const mySsm = new ssm.StringParameter(this, 'S3BucketNameSSMParam', {
-      parameterName: "AB-S3-ConfigBucket",
-
-      description: 'S3 bucket used to store config for AB testing workshop',
-      stringValue: configBucket.bucketName
-    });
-
-
-    new s3deployment.BucketDeployment(this, "deployment", {
-      sources: [s3deployment.Source.asset("./resources/website")],
+    const hostingBucket = new Bucket(this, "bucket");
+    new BucketDeployment(this, "deployment", {
+      sources: [Source.asset("./resources/website")],
       destinationBucket: hostingBucket,
     });
 
-    new s3deployment.BucketDeployment(this, "config", {
-      sources: [s3deployment.Source.asset("resources/module_3_1/segmentation")],
-      destinationBucket: configBucket,
+    const viewerRequest = new FunctionWithStore(this, "viewer-request", {
+      entryPath: RESOURCES_PATH_PREFIX + "/viewer-request-function.js",
+      keyValueStoreImportSourcePath: RESOURCES_PATH_PREFIX + "/viewer-request-store-source.json"
     });
 
-    const lambdaEdgeViewerRequest = new lambda.Function(this, 'LambdaEdgeViewerRequest', {
-      runtime: lambda.Runtime.NODEJS_16_X,
-      handler: 'index.handler',
-      code: lambda.Code.fromAsset('resources/module_3_1/request'),
-    });
+    const viewerResponse = new Function(this, "viewer-response", {
+      code: FunctionCode.fromFile({ filePath: RESOURCES_PATH_PREFIX + "/viewer-request-function.js" }),
```

Comment:
viewer response points to viewer request code

Reply:
Fixed in https://github.com/piotrekwitkowski/ab-testing-at-edge/commit/672897e4d00bcd483e0d63ce904ff8208a2f4f68